### PR TITLE
Delete unpublished versions and history in package.json, fixes #1359

### LINKED ISSUE #1359: After unpublish a package，I can't publish the package
version: 4.0.2

I'm sorry, my English is poor.
Now I try to describe my problem in english, 
I hope it helps you understand what I am talking about.

**Describe the bug**
I published a package twice(version 1.0.0 & version 1.0.1)，after some minutes I unpublished the package which version was 1.0.1 by used the commend "npm unpublish --force".
No problem until this step.

But when I tried to publish the package which version was 1.0.1 once again, it didn't work。
It told me that I need to update my version, but you know, the version 1.0.1 has been unpublish successly.

Then I got into the path which storing the package file，I found the package file(.tgz) had been deleted actually. But the file package.json in the same path didn't be changed, it still record the information of version 1.0.1.

**To Reproduce**
1. Publish a package which version is 1.0.0.
2. Update the version of the package to 1.0.1 and publish it.
3. Unpublish the package which version 1.0.1 by commend ”npm unpublish --force".
3. Publish the package which version 1.0.1 once again.
4. See error.

I wish I had made it clear.

### PR DESCRIPTION
**Pull request by AllenBird**
** Type: bug **

** Description: ** 
Delete versions and history in package.json, to fix issue #1359.
https://github.com/verdaccio/verdaccio/issues/1359

### CODE REVIEW

**Comment by DanielRuf** [comment]
Is this relevant?

> I think it should be an option in verdaccio

If so please make it configurable. Or does it fix another issue which is not about the initial comment in the linked issue?

**Comment by DanielRuf** [comment]
Can you add some unit tests which fail without your changes and succeed with your changes?

File: src/lib/local-storage.js [hunk 1]
```diff
@@ -360,6 +360,15 @@ class LocalStorage implements IStorage {
       name,
       (data, cb) => {
         if (data._attachments[filename]) {
+          // get version form _attachments;
```
Comment (DanielRuf) [inline src/lib/local-storage.js]: form => from

File: src/lib/local-storage.js [hunk 2]
```diff
@@ -360,6 +360,15 @@ class LocalStorage implements IStorage {
...
+          // there aways had 'version' in verdaccio publish package.
```
Comment (DanielRuf) [inline src/lib/local-storage.js]: Not sure what you mean here.

File: src/lib/local-storage.js [hunk 3]
```diff
@@ -360,6 +360,15 @@ class LocalStorage implements IStorage {
...
+          // but there was no 'version' in package witch unlink form npmjs
```
Comment (DanielRuf) [inline src/lib/local-storage.js]: form => from
witch => which

What is this line about?

Comment (AllenBird) [reply src/lib/local-storage.js]: // [1] get version from _attachments;
// [2] there aways had 'version' in data._attachments[filename] which developer publish the package to verdaccio.
// [3] but there was no 'version' in data._attachments[filename], if the package uplink(pull) form source registry like 'www.npmjs.com'
 
for example: 
I fork react 16.0.0 and make a new version called 16.1.0.
then I publish it to my private verdaccio.
verdaccio will pull react from source registry.
If I unpublish react@16.1.0, it's the case above [2]
If I unpublish react@16.0.0, it's the case above [3]

**Comment by AllenBird** [comment]
> Is this relevant?
> 
> > I think it should be an option in verdaccio
> 
> If so please make it configurable. Or does it fix another issue which is not about the initial comment in the linked issue?

I think it's part of the design.maybe we can discuss: 
1. when to complete the configuration 
2. which is the default way

**Comment by juanpicado** [comment]
@AllenBird, thanks for the PR, looks promising, this fix requires a unit test or update the current ones, you can find more info about it here https://github.com/verdaccio/verdaccio/blob/master/test/README.md or ask here for any mentoring, I'll be happy to help.

File: src/lib/local-storage.js (hunk 3 above)
Comment (DanielRuf) [reply src/lib/local-storage.js]: I think we should completely remove these comments (they are too verbose) and only describe that we remove the relevant parts on `unpublish`.

**Comment by AllenBird** [comment]
I find that this solution introduces problems.
When developer delete the latest version, All version cannot be displayed normally.
I will found another solution.

**Pull request closed without merging**